File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/robotcrosswords/data/RobotCrosswordsDocument.java
package com.zwstudio.logicpuzzlesandroid.puzzles.robotcrosswords.data;

import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument;
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress;
import com.zwstudio.logicpuzzlesandroid.common.domain.Position;
import com.zwstudio.logicpuzzlesandroid.puzzles.robotcrosswords.domain.RobotCrosswordsGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.robotcrosswords.domain.RobotCrosswordsGameMove;

import org.androidannotations.annotations.EBean;

@EBean
public class RobotCrosswordsDocument extends GameDocument<RobotCrosswordsGame, RobotCrosswordsGameMove> {
    protected void saveMove(RobotCrosswordsGameMove move, MoveProgress rec) {
        rec.row = move.p.row;
        rec.col = move.p.col;
        rec.intValue1 = move.obj;
    }
    public RobotCrosswordsGameMove loadMove(MoveProgress rec) {
        return new RobotCrosswordsGameMove() {{
            p = new Position(rec.row, rec.col);
            obj = rec.intValue1;
        }};
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/robotcrosswords/RobotCrosswordsDocument.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.robotcrosswords

import android.content.Context
import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress
import com.zwstudio.logicpuzzlesandroid.common.domain.Position

class RobotCrosswordsDocument(context: Context) : GameDocument<RobotCrosswordsGameMove>(context) {
    override fun saveMove(move: RobotCrosswordsGameMove, rec: MoveProgress) {
        rec.row = move.p.row
        rec.col = move.p.col
        rec.intValue1 = move.obj
    }

    override fun loadMove(rec: MoveProgress) =
        RobotCrosswordsGameMove(Position(rec.row, rec.col), rec.intValue1)
}